docs/styleguide/PyGuide.md

```diff
@@ -719,21 +719,62 @@ long_string = textwrap.dedent('''\
 <a id="37-logging"></a>
 <a id="logging"></a>
 ### 3.7 Logging 
+Use the logger object built into NNCF for all purposes of logging within the NNCF package code.
+Do not use `print(...)` or other ways of output.
+
+Correct:
+
+```python
+from nncf.common.logging import nncf_logger
+
+nncf_logger.info("This is an info-level log message")
+```
+
+Wrong:
+```python
+print("This is an info-level log message")
+```
 
 For logging functions that expect a pattern-string (with %-placeholders) as
-their first argument: Always call them with a string literal (not an f-string!)
-as their first argument with pattern-parameters as subsequent arguments. Some
-logging implementations collect the unexpanded pattern-string as a queryable
-field. It also prevents spending time rendering a message that no logger is
-configured to output.
+their first argument - consider calling them with a string literal (not an f-string!)
```

Comment:
I don't think that it is a good point and where we should guide and limit ourselves. I also wonder where it comes from. Some comparisons do not show significant gaps, e.g. https://blog.finxter.com/string-formatting-vs-format-vs-formatted-string-literal
I personally love f-strings and use them only.

Reply:
https://stackoverflow.com/questions/54367975/python-3-7-logging-f-strings

I used the "consider" part and the "if" clause in the end of the sentence specifically to make sure that this recommendation is to be applied only where it is really necessary.